### PR TITLE
community: add `ChatSnowflakeCortex` chat model

### PR DESCRIPTION
**Pull request by sp35**
**Description:** This PR adds a chat model integration for [Snowflake Cortex](https://docs.snowflake.com/en/user-guide/snowflake-cortex/llm-functions), which gives an instant access to industry-leading large language models (LLMs) trained by researchers at companies like Mistral, Reka, Meta, and Google, including [Snowflake Arctic](https://www.snowflake.com/en/data-cloud/arctic/), an open enterprise-grade model developed by Snowflake.

**Dependencies:** Snowflake's [snowpark](https://pypi.org/project/snowflake-snowpark-python/) library is required for using this integration.

**Twitter handle:** [@gethouseware](https://twitter.com/gethouseware)

- [x] **Add tests and docs**:
  1. integration tests: `libs/community/tests/integration_tests/chat_models/test_snowflake.py`
  2. unit tests: `libs/community/tests/unit_tests/chat_models/test_snowflake.py`
  3. example notebook: `docs/docs/integrations/chat/snowflake.ipynb`

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

### CODE REVIEW

**Comment by AryanMadhavVerma** [comment]
🚀

**Comment by sp35** [comment]
Bumping this up, @baskaryan can you please review this?

**Comment by sp35** [comment]
@eyurtsev, would love to hear your feedback on this or anything I can do to make this available to Langchain community.

**Review by baskaryan — APPROVED** [review]

**Comment by baskaryan** [comment]
spellcheck CI should be fixed if you merge in latest master. also happy to help land if you want to give me write access to your fork!

**Review by baskaryan — APPROVED** [review]

File: docs/docs/integrations/chat/snowflake.ipynb
Comment (baskaryan) [inline docs/docs/integrations/chat/snowflake.ipynb:1]: title shoud have single # and can be the name of the underlying tech, doesn't have to class name:
```markdown
# Snowflake Cortex
```

**Review by baskaryan — COMMENTED** [review]
you'll want to run 
```
poetry install --with lint
make format
```
from root dir to format the docs correctly. if you give me write access to your fork happy to help as well!

File: docs/docs/integrations/chat/snowflake.ipynb
Comment (sp35) [reply docs/docs/integrations/chat/snowflake.ipynb:1]: Makes sense, added this change!

**Pull request merged**